src/libcore/iter/traits.rs

```diff
@@ -798,6 +798,21 @@ impl<I, T, E> Iterator for ResultShunt<I, E>
 impl<T, U, E> Sum<Result<U, E>> for Result<T, E>
     where T: Sum<U>,
 {
+    /// Takes each element in the `Iterator`: if it is an `Err`, no further
+    /// elements are taken, and the `Err` is returned. Should no `Err` occur,
+    /// the sum of all elements is returned.
+    ///
+    /// Here is an example which sums up every integer in a vector,
```

Comment:
As per https://github.com/rust-lang/rfcs/pull/1574, there should be an `# Examples` line before this line